privaterelay/middleware.py

```diff
@@ -39,7 +39,7 @@ def __call__(self, request):
                     FirefoxAccountsOAuth2Adapter.access_token_url
                 )
                 update_social_token(social_token, new_token)
-            except CustomOAuth2Error:
+            except (CustomOAuth2Error, NoSocialToken):
```

Comment:
Nice! Catching and handling for tokens that are invalid/dne during user interaction! TIL that middlewares are only applied when loading views (see this [doc](https://docs.djangoproject.com/en/3.2/topics/http/middleware/#middleware-order-and-layering)).